src/ai/chunking/ai-chunking-agentic.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  AzureKeyCredential,
+  type ChatRequestMessageUnion,
+  OpenAIClient,
+} from '@azure/openai';
+// eslint-disable-next-line camelcase
+import { get_encoding } from 'tiktoken';
+import { createDocumentChunks } from './ai-chunking-fixed-size.js';
+
+if (process.env.OPENAI_URL === undefined) {
+  throw new Error('please set the env variable OPENAI_URL');
+}
+if (process.env.OPENAI_KEY === undefined) {
+  throw new Error('please set the env variable OPENAI_KEY');
+}
+
+const modelName = 'gpt-35-turbo-16k'; // gpt-35-turbo version 1106.
```

Comment:
no need to specify the model name, put this directly on the model call since you call it only once